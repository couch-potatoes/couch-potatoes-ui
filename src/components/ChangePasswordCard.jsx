import React from 'react';
import { connect } from 'react-redux';
import {
  Card,
  RaisedButton,
  TextField,
} from 'material-ui';

import { changeUserPassword } from '../actions/auth';
import { addNotification } from '../actions/notifications';

const styles = {
  paper: {
    display: 'flex',
    flexFlow: 'column wrap',
    alignContent: 'space-around',
    padding: '2rem',
  },
};

class ChangePasswordCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPassword: '',
      newPassword: '',
      newPasswordAgain: '',
    };
    this.handleTextFieldChange = this.handleTextFieldChange.bind(this);
    this.changePassword = this.changePassword.bind(this);
  }

  handleTextFieldChange(event) {
    const {
      target: {
        name,
        value,
      },
    } = event;
    this.setState({
      [name]: value,
    });
  }

  changePassword() {
    const {
      currentPassword,
      newPassword,
      newPasswordAgain,
    } = this.state;
    const {
      dispatch,
    } = this.props;
    if (newPassword !== newPasswordAgain) {
      dispatch(addNotification('Passwords do not match'));
      return;
    }
    dispatch(changeUserPassword(currentPassword, newPassword))
      .then(() => {
        this.setState({
          currentPassword,
          newPassword,
          newPasswordAgain,
        });
      });
  }

  render() {
    const {
      currentPassword,
      newPassword,
      newPasswordAgain,
    } = this.state;
    return (
      <div>
        <Card style={styles.paper}>
          <TextField
            fullWidth
            floatingLabelText="Current Password"
            name="currentPassword"
            onChange={this.handleTextFieldChange}
            spellCheck={false}
            type="password"
            value={currentPassword}
          />
          <TextField
            fullWidth
            floatingLabelText="New Password"
            name="newPassword"
            onChange={this.handleTextFieldChange}
            spellCheck={false}
            type="password"
            value={newPassword}
          />
          <TextField
            fullWidth
            floatingLabelText="Confirm New Password"
            name="newPasswordAgain"
            onChange={this.handleTextFieldChange}
            spellCheck={false}
            type="password"
            value={newPasswordAgain}
          />
        </Card>
        <RaisedButton
          className="inline-button"
          label="Submit"
          onTouchTap={this.changePassword}
        />
      </div>
    );
  }
}

export default connect()(ChangePasswordCard);
