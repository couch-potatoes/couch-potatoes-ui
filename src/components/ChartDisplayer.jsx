import React, { PropTypes } from 'react';
import { Chart } from 'react-google-charts';

const lineChartOptions = {
  title: 'Daily Nutritional Statistics as Reported by Students',
  hAxis: { title: 'Date' },
  vAxis: { title: 'Amount' },
  legend: 'all',
};

const barChartOptions = {
  title: 'Daily Nutritional Statistics as Reported by Students',
  vAxis: { title: 'Amount' },
  hAxis: { title: 'Date' },
  legend: 'all',
};

const ChartDisplayer = ({ chartData }) => {
  if (chartData.length) {
    return (
      <div>
        <Chart
          chartType="BarChart"
          data={chartData}
          options={barChartOptions}
          graph_id="BarChart"
          width="100%"
          height="600px"
          legend_toggle
        />
        <Chart
          chartType="LineChart"
          data={chartData}
          options={lineChartOptions}
          graph_id="LineChart"
          width="100%"
          height="600px"
          legend_toggle
        />
      </div>
    );
  }
  // Render nothing if there is no data to display
  return null;
};

ChartDisplayer.propTypes = {
  chartData: PropTypes.array.isRequired,
};

export default ChartDisplayer;
