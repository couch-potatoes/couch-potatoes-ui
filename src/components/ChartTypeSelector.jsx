import React, {PropTypes} from 'react';
import {
  MenuItem,
  SelectField,
} from 'material-ui';

const ChartTypeSelector = ({ onChange, value }) => (
  <SelectField
    fullWidth
    floatingLabelText="Chart Type"
    id="chart-type"
    onChange={onChange}
    value={value}
  >
    <MenuItem value="" primaryText="" />
    <MenuItem value="nutrition" primaryText="Nutrition" />
    <MenuItem value="wellness" primaryText="Wellness" />
    <MenuItem value="fitness" primaryText="Fitness" />
  </SelectField>
);

ChartTypeSelector.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default ChartTypeSelector;
