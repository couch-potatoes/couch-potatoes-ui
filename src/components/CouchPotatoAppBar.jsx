import React, { Component, PropTypes } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { AppBar } from 'material-ui';

import Login from './buttons/Login';
import Menu from './buttons/Menu';

import { logout } from '../actions/auth';

class CouchPotatoAppBar extends Component {
  constructor(props) {
    super(props);
    this.routeToHome = this.routeToHome.bind(this);
    this.routeToLogin = this.routeToLogin.bind(this);
    this.signOut = this.signOut.bind(this);
  }
  routeToHome() {
    const { history } = this.props;
    history.push('/');
  }
  routeToLogin() {
    const { history } = this.props;
    history.push('/login');
  }
  signOut() {
    const { dispatch, history } = this.props;
    dispatch(logout())
      .then(() => { history.push('/login'); });
  }
  render() {
    const {
      isLoggedIn,
    } = this.props;
    let iconElementRight;
    if (isLoggedIn) {
      iconElementRight = <Menu signOut={this.signOut} />;
    } else {
      iconElementRight = <Login loginCallback={this.routeToLogin} />;
    }
    return (
      <div>
        <AppBar
          title="Athlete Tracker Program"
          iconElementRight={iconElementRight}
          onTitleTouchTap={this.routeToHome}
          showMenuIconButton={false}
        />
      </div>
    );
  }
}

CouchPotatoAppBar.propTypes = {
  dispatch: PropTypes.func.isRequired,
  isLoggedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => {
  const {
    account: {
      isLoggedIn,
    },
  } = state;
  return {
    isLoggedIn,
  };
};

export default withRouter(connect(mapStateToProps)(CouchPotatoAppBar));
