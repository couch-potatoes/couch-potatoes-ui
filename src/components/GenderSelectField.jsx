import React, { PropTypes } from 'react';
import {
  MenuItem,
  SelectField,
} from 'material-ui';

    //function to handle when selecting genders
const GenderSelector = (props) => {
  const { disabled, onChange, value } = props;
  return (
    <SelectField
      disabled={disabled}
      floatingLabelText="Gender"
      fullWidth
      id="gender"
      onChange={onChange}
      value={value}
    >
      <MenuItem value='' primaryText="" />
      <MenuItem value="male" primaryText="Male" />
      <MenuItem value="female" primaryText="Female" />
    </SelectField>
  );
};


GenderSelector.propTypes = {
  disabled: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default GenderSelector;
