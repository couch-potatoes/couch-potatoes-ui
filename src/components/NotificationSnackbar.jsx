import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import Snackbar from 'material-ui/Snackbar';

import { closeNotification } from '../actions/notifications';

const NotificationSnackbar = (props) => {
  const {
    closeNotification,
    displayed,
    message,
  } = props;

  return (
    <Snackbar
      open={displayed}
      message={message}
      autoHideDuration={5000}
      onRequestClose={closeNotification}
    />
  );
};

NotificationSnackbar.propTypes = {
  closeNotification: PropTypes.func.isRequired,
  displayed: PropTypes.bool.isRequired,
  message: PropTypes.string,
};

NotificationSnackbar.defaultProps = {
  message: '',
};

const mapDispatchToProps = (dispatch) => ({
  closeNotification: () => { dispatch(closeNotification()); }
});

const mapStateToProps = (state) => {
  const {
    notifications: {
      displayed,
      messages,
    },
  } = state;
  return {
    displayed,
    message: messages[0],
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(NotificationSnackbar);
