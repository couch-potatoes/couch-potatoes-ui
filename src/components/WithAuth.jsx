import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

class WithAuth extends React.Component {
  render() {
    const {
      children,
      isLoggedIn,
    } = this.props;
    if (!isLoggedIn) {
      // Redirect user to login page
      return (
        <Redirect to="login"/>
      );
    }
    return children;
  }
}

const mapStateToProps = (state) => {
  const {
    auth: {
      isLoggedIn,
    },
  } = state;

  return {
    isLoggedIn,
  };
}

export default connect(mapStateToProps)(WithAuth);
