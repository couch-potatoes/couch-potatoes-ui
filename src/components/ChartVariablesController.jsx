import React, { PropTypes } from 'react';

import {
  Card,
  DatePicker,
} from 'material-ui';

import ChartTypeSelector from './ChartTypeSelector';
import GenderSelectField from './GenderSelectField';
import SportSelectField from './SportSelectField';
import dates from '../util/dates';

const ChartVariablesController = (props) => {
  const {
    handleChartTypeChange,
    handleDateChange,
    handleGenderFieldChange,
    handleSportsFieldChange,
    chartQueryData: {
      chartType,
      endDate,
      gender,
      sports,
      startDate,
    },
  } = props;

  return (
    <Card>
      <ChartTypeSelector
        onChange={handleChartTypeChange}
        value={chartType}
      />
      <DatePicker
        floatingLabelText="Start of Range Date"
        name="startDate"
        container="inline"
        locale="en-US"
        mode="landscape"
        minDate={dates.minDate}
        maxDate={dates.currentDate}
        value={startDate}
        onChange={handleDateChange('startDate')}
      />
      <DatePicker
        floatingLabelText="End of Date Range"
        name="endDate"
        container="inline"
        locale="en-US"
        mode="landscape"
        value={endDate}
        minDate={startDate}
        maxDate={dates.currentDate}
        onChange={handleDateChange('endDate')}
      />
      <GenderSelectField
        disabled={false}
        onChange={handleGenderFieldChange}
        value={gender}
      />
      <SportSelectField
        disabled={false}
        onChange={handleSportsFieldChange}
        gender={gender}
        value={sports}
      />
    </Card>
  );
};

ChartVariablesController.propTypes = {
  chartQueryData: PropTypes.shape({
    startDate: PropTypes.date,
    endDate: PropTypes.date,
    gender: PropTypes.string,
    sports: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
  handleChartTypeChange: PropTypes.func.isRequired,
  handleDateChange: PropTypes.func.isRequired,
  handleGenderFieldChange: PropTypes.func.isRequired,
  handleSportsFieldChange: PropTypes.func.isRequired,
};

export default ChartVariablesController;
