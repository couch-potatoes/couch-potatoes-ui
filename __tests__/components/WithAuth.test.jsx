import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import { WithAuth } from '../../src/components/WithAuth';

describe('<WithAuth />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot when user is logged in', () => {
    const wrapper = shallowWithContext(
      <WithAuth
        children={<h1>foo</h1>}
        isLoggedIn={true}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('matches snapshot when user is not logged in', () => {
    const wrapper = shallowWithContext(
      <WithAuth
        children={<h1>bar</h1>}
        isLoggedIn={false}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  describe('prop: isLoggedIn', () => {
    it('should render the children if false', () => {
      const children = <h1>bar</h1>;
      const wrapper = shallowWithContext(
        <WithAuth
          children={children}
          isLoggedIn={true}
        />
      );
      const childElement = wrapper.find('h1');
      expect(childElement.length).toBe(1);
    });
  });
});
