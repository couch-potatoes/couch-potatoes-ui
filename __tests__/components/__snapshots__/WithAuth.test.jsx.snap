exports[`<WithAuth /> matches snapshot when user is logged in 1`] = `
<h1>
  foo
</h1>
`;

exports[`<WithAuth /> matches snapshot when user is not logged in 1`] = `
<Redirect
  push={false}
  to="login" />
`;
